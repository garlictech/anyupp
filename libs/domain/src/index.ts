export * from './lib/use-cases';
export * from './lib/entities';
