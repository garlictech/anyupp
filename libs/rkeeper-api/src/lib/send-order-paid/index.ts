export { sendOrderPaid } from './send-order-paid';
