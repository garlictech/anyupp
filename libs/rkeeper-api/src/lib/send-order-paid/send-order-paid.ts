import { AxiosStatic } from 'axios';
import { RKeeper } from '@bgap/domain';

interface Params {
  axiosInstance: AxiosStatic;
  rKeeperPosConfig: RKeeper;
  restaurantId: string;
  visitId: string;
}

export const sendOrderPaid = (params: Params): Promise<unknown> => {
  const { axiosInstance, rKeeperPosConfig, restaurantId, visitId } = params;

  return axiosInstance.request({
    url: `${rKeeperPosConfig.endpointUri}/postorder/payed/${restaurantId}/${visitId}`,
    method: 'post',
    auth: {
      username: rKeeperPosConfig.rkeeperUsername,
      password: rKeeperPosConfig.rkeeperPassword,
    },
  });
};
