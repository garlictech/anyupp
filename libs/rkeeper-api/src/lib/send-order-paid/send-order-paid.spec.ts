import { RKeeper } from '@bgap/domain';
import { sendOrderPaid } from './send-order-paid';
import { AxiosStatic } from 'axios';

describe('SendOrderPaid', () => {
  const mockRestaurantId = '109150001';
  const mockVisitId = '1656509181-606-109150001';
  const mockRKeeperPos: RKeeper = {
    endpointUri: 'https://testendpoint.ucs.hu/wp-json/vendor/v1',
    rkeeperUsername: '795_50_155_539',
    rkeeperPassword: 'b6302d53085c9486d0f765ec475f18',
    anyuppUsername: 'user',
    anyuppPassword: 'Ch4nge1t!',
  };

  let mockAxiosInstance: { request: jest.Mock };

  beforeEach(() => {
    mockAxiosInstance = {
      request: jest.fn().mockReturnValue(Promise.resolve({})),
    };
  });

  it('should call correct URI - happy path', async () => {
    await sendOrderPaid({
      axiosInstance: mockAxiosInstance as unknown as AxiosStatic,
      restaurantId: mockRestaurantId,
      visitId: mockVisitId,
      rKeeperPosConfig: mockRKeeperPos,
    });

    expect(mockAxiosInstance.request.mock.calls).toMatchSnapshot();
  });
});
