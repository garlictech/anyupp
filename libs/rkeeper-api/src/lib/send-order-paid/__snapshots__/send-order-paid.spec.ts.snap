// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendOrderPaid should call correct URI - happy path 1`] = `
Array [
  Array [
    Object {
      "auth": Object {
        "password": "RKEEPER_PASSWORD",
        "username": "RKEEPER_USERNAME",
      },
      "method": "post",
      "url": "RKEEPER_ENDPOINT/postorder/payed/109150001/1656509181-606-109150001",
    },
  ],
]
`;
